File: src/BHS.Web/ClientApp/src/app/data/schema/post.ts
import { Author } from './author';
import { Category } from './category';

export interface Post {
  slug: string;
  title: string;
  contentMarkdown: string;
  filePath: string;
  photosAlbumId?: number;
  author?: Author;
  datePublished: Date;
  dateLastModified: Date;
  categories: Category[];
}

File: src/BHS.Contracts/Blog/Post.cs
namespace BHS.Contracts.Blog
{
    public record Post(
        string Slug,
        string Title,
        string ContentMarkdown,
        Uri? FilePath,
        string? PhotosAlbumSlug,
        Author? Author,
        DateTimeOffset DatePublished,
        DateTimeOffset DateLastModified,
        ICollection<Category> Categories);
}
